core/src/saros/session/User.java

```diff
@@ -54,7 +55,74 @@
 
   private final int favoriteColorID;
 
+  @Deprecated
   private volatile Permission permission = Permission.WRITE_ACCESS;
+  
+  
+  
+  
+  
+  
+  /* More flexible Permissions also known as Privilege */
+  private Map<UserPrivilege.Keys, UserPrivilege> privileges;
+  public Map<UserPrivilege.Keys, UserPrivilege> getPrivileges() {
+      return this.privileges;
+  }
+  public void setPrivileges(Map<UserPrivilege.Keys, UserPrivilege> privileges) {
+      this.privileges = privileges;
+  }
+  public void addPrivilege(UserPrivilege privilege) {
+      this.privileges.put(privilege.getKey(), privilege);
+  }
+  // get any privileges value or false
+  public boolean hasPrivilege(UserPrivilege.Keys privilege) {
+//System.out.println("3 - hasPrivilege() " + privilege + " : " + this.privileges.containsKey(privilege));
+	  if (this.privileges.containsKey(privilege)) {
+		  return this.privileges.get(privilege).getValue();
+	  }
+	  return false;
+  }
+
+  // convenience functions to privilege values
+  public boolean hasReadOnlyAccessPrivilege() {
```

Comment:
I do not see the need for 1 line convenience methods. Using the enum Name should be sufficient.